controllers/ocsinitialization/ocsinitialization_controller.go

```diff
@@ -253,13 +253,13 @@ func (r *OCSInitializationReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	)
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&ocsv1.OCSInitialization{}).
-		Owns(&corev1.Service{}).
-		Owns(&corev1.Secret{}).
-		Owns(&promv1.Prometheus{}).
-		Owns(&corev1.ConfigMap{}).
-		Owns(&promv1.Alertmanager{}).
-		Owns(&promv1.ServiceMonitor{}).
+		For(&ocsv1.OCSInitialization{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Owns(&corev1.Service{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Owns(&corev1.Secret{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Owns(&promv1.Prometheus{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Owns(&corev1.ConfigMap{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Owns(&promv1.Alertmanager{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Owns(&promv1.ServiceMonitor{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
```

Comment:
@iamniting I am not sure generation will change for metadata changes, like labels and annotation (I am quite sure the opposite is correct)

Reply:
```
// GenerationChangedPredicate implements a default update predicate function on Generation change.                                                                                            
//                                                                              
// This predicate will skip update events that have no change in the object's metadata.generation field.
// The metadata.generation field of an object is incremented by the API server when writes are made to the spec field of an object.
// This allows a controller to ignore update events where the spec is unchanged, and only the metadata and/or status fields are changed.
//                                                                              
// For CustomResource objects the Generation is only incremented when the status subresource is enabled.
//                                                                              
// Caveats:                                                                     
//                                                                              
// * The assumption that the Generation is incremented only on writing to the spec does not hold for all APIs.
// E.g For Deployment objects the Generation is also incremented on writes to the metadata.annotations field.
// For object types other than CustomResources be sure to verify which fields will trigger a Generation increment when they are written to.
//                                                                              
// * With this predicate, any update events with writes only to the status field will not be reconciled.
// So in the event that the status block is overwritten or wiped by someone else the controller will not self-correct to restore the correct status.
```

This is a comment from the pkg.

Reply:
Yes, the above proves my point. 
``` This predicate will skip update events that have no change in the object's metadata.generation field.
// The metadata.generation field of an object is incremented by the API server when writes are made to the spec field of an object.
// This allows a controller to ignore update events where the spec is unchanged, and only the metadata and/or status fields are changed.```

Reply:
Why do we care about the status and metadata of all the objects?

We are worried about the ceph cluster and noobaa status and for those, I have not added GenerationChangedPredicate.

Reply:
Because there are resources where our reconciliation adds labels/annotations and we want an event whenever these are touched. Trying to track which one we do and which we don't, using a different code path (using the annotation vs not using the annotation) will fail very fast when any other developer adds an annotation to an existing resource (as part of some current or future change)  but will not know that they need to change the Watch!

Reply:
I removed `GenerationChangedPredicate`, so we should be good now.